src/test/ui/issue-40402-ref-hints/issue-40402-1.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
```

Comment:
It'd be good to put a comment at the top of this file explaining the purpose of the test:

```
// Check that we do not suggest `ref f` here in the `main()` function.
```